### PR TITLE
Grafana UI: Bump to Storybook 8

### PR DESCRIPTION
**Pull request by jackw**
**What is this feature?**

This PR updates Grafana UI to use Storybook 8.0.x.

**Why do we need this feature?**

To bring the new and shiny to our component library.

**Who is this feature for?**

Grafana contributors

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

We're continuing to patch the export of `<Preview />` in storybook/blocks.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Layout/Box/Box.mdx [hunk 1]
```diff
@@ -32,7 +32,7 @@ If you need layout styles, use the Stack or Grid components instead.
         [cssProp]: theme.spacing(val),
     })),
    ```
-4. Add it to the `Box` story in `Box.internal.story.tsx`, by explicity adding it to `Basic.argTypes`
+4. Add it to the `Box` story in `Box.internal.story.tsx`, by explicity adding it to `Basic.ArgTypes`
```
Comment (ashharrison90) [inline packages/grafana-ui/src/components/Layout/Box/Box.mdx:35]: we sure we wanna change the casing here? the casing in the story is `argTypes`

**Review by ashharrison90 — COMMENTED** [review]
amazing work @jackw 👏 

i'll try take a look at the `iconOptions` typing tomorrow as well

File: packages/grafana-ui/src/components/Layout/Box/Box.mdx (hunk 1 above)
Comment (jackw) [reply packages/grafana-ui/src/components/Layout/Box/Box.mdx]: Nope. Reverted this. Thanks!

File: packages/grafana-ui/src/utils/storybook/themeStorybookControls.tsx [hunk 2]
```diff
@@ -1 +1,6 @@
-export const SpacingTokenControl = { control: 'select', options: [0, 0.25, 0.5, 1, 1.5, 2, 3, 4, 5, 6, 8, 10] };
+import { Control } from '@storybook/addon-docs';
+
+export const SpacingTokenControl = {
+  control: 'select' as Control,
```
[the remark this answers is not in the document]
Comment (jackw) [reply packages/grafana-ui/src/utils/storybook/themeStorybookControls.tsx]: Ah nice! Didn't think of that, tis updated.

**Review by ashharrison90 — APPROVED** [review]
looks great! 🚀🚀🚀🚀🚀

@JoaoSilvaGrafana do you wanna take a look as well since i think you're more up-to-date on the state of storybook than me 🙏

**Review by JoaoSilvaGrafana — APPROVED** [review]
Great job on this Jack 🚀 

I had a play around and it all looks like its working!

**Pull request merged**